src/unity/python/turicreate/data_structures/sframe.py

```diff
@@ -3520,7 +3517,9 @@ def __getitem__(self, key):
         """
         if type(key) is SArray:
             return self._row_selector(key)
-        elif type(key) is str:
+        elif isinstance(key, six.string_types):
```

Comment:
Doesn't this also allow `bytes` access in Python 3? I am fine with dual bytes/unicode indexing in Python 2 since they coerce implicitly. However, in Python 3 (where they don't, i.e. `b'foo' != u'foo'`), dual indexing can cause confusion and would be inconsistent with Python 3 semantics.

Consider this situation in Python 3:

```python
sf = tc.SFrame({'foo': [{'bar': 100}]})
sf['foo'][0]  # returns Python dictionary
sf['foo'][0]['bar']  # Returns 100

sf[b'foo']  # This would work if we allow bytes indexing
sf[0][b'foo']  # We could allow this too if extended to SArray
sf[0][b'foo'][b'bar']  # This will never work, since we can't control the semantics of native Python types
```
I think it's better to let SFrame/SArray follow Python 3 semantics and not mix.

Reply:
Never mind, I misremembered what `six.string_types` gives you. This gives the desired semantics already.